### PR TITLE
Dynamically get trust domain from the cluster for telemetry test.

### PR DESCRIPTION
**Pull request by bianpengyuan**
[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: tests/integration/telemetry/util.go [hunk 1]
```diff
@@ -39,3 +43,24 @@ func PromDumpWithAttributes(cluster cluster.Cluster, prometheus prometheus.Insta
...
+	if err != nil {
+		return defaultTrustDomain
+	}
+
+	configYaml, ok := meshConfigMap.Data["mesh"]
+	if !ok {
+		return defaultTrustDomain
+	}
+
+	cfg, err := mesh.ApplyMeshConfigDefaults(configYaml)
```
Comment (richardwxn) [inline tests/integration/telemetry/util.go:60]: The ApplyMeshConfigDefaults function may just override the one you returned from the cluster with default value, i.e. trustDomain=cluster.local here?

Comment (bianpengyuan) [reply tests/integration/telemetry/util.go:60]: Default mesh config is applied on fields which is not specified.

**Review by richardwxn — APPROVED** [review]

**Comment by bianpengyuan** [comment]
@istio/wg-policies-and-telemetry-maintainers ptal, thanks!

**Comment by bianpengyuan** [comment]
@istio/wg-policies-and-telemetry-maintainers ptal, thanks!

**Comment by bianpengyuan** [comment]
@istio/wg-policies-and-telemetry-maintainers ptal, thanks!

**Comment by bianpengyuan** [comment]
Ping @istio/wg-policies-and-telemetry-maintainers ptal thanks!

**Review by kyessenov — APPROVED** [review]

**Pull request merged**